libs/ui/src/components/SignupForm/SignupForm.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react'
+
+import {
+  Button,
+  Checkbox,
+  Container,
+  Divider,
+  Heading,
+  HStack,
+  Stack,
+  Text,
+} from '@chakra-ui/react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useTranslation } from 'next-i18next'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { TFunction } from 'react-i18next'
+import * as yup from 'yup'
+
+import { FormItem } from '../FormItem'
+import { Navigate } from '../Navigate'
+import { OAuthButtonGroup } from '../OauthButtonGroup'
+import { SignupFormProps, SignupFormFieldValues } from './types'
+
+const schema = (t: TFunction<'translation', undefined>) =>
+  yup.object({
+    name: yup.string().required(t('login.name.required')),
+    username: yup.string().required(t`login.username.required`),
+    password: yup
+      .string()
+      .min(8, t('login.password.warning'))
+      .required(t('login.password.required'))
+      .matches(RegExp('(.*[a-z].*)'), t('login.password.matches.lowercase'))
+      .matches(RegExp('(.*[A-Z].*)'), t('login.password.matches.uppercase'))
+      .matches(RegExp('(.*\\d.*)'), t('login.password.matches.number'))
+      .matches(
+        RegExp('[!@#$%^&*(),.?":{}|<>]'),
+        t('login.password.matches.special'),
+      ),
+    email: yup
+      .string()
+      .email(t`contact.form.email-invalid`)
+      .required(t`login.email.required`),
+  })
+
+export const SignupForm: React.FC<SignupFormProps> = ({
+  onhandleSubmitSignUp,
+  handleTermsAccepted,

```

Comment:
```suggestion
  onSignup,
  onAcceptTerms,
```

We can use shorter descriptive prop names for the components.